onnx/test/shape_inference_test.py

```diff
@@ -3455,6 +3455,39 @@ def test_trilu_lower_one(self):  # type: () -> None
             [],)
         self._assert_inferred(graph, [make_tensor_value_info('y', TensorProto.INT32, (3, 1, 5))])  # type: ignore
 
+    def test_batch_norm_train(self):  # type: () -> None
+        graph = self._make_graph(
+            [('x', TensorProto.FLOAT, (3, 4, 5, 6, 7)),
+             ('scale', TensorProto.FLOAT, (4,)),
+             ('b', TensorProto.FLOAT, (4,)),
+             ('mean', TensorProto.FLOAT, (4,)),
+             ('var', TensorProto.FLOAT, (4,)),
+             ('training_mode', TensorProto.BOOL, ())],
```

Comment:
How about adding one test without the optional training_mode input ?

Reply:
Such a test is there in line 1239 for batch_norm.